libbeat/processors/actions/extract_field_test.go

```diff
@@ -0,0 +1,87 @@
+package actions
+
+import (
+	"testing"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCommonPaths(t *testing.T) {
+	var tests = []struct {
+		Value, Field, Separator, Target, Result string
+		Index                                   int
+	}{
+		// Common docker case
+		{
+			Value:     "/var/lib/docker/containers/f1510836197d7c34da22cf796dba5640f87c04de5c95cf0adc11b85f1e1c1528/f1510836197d7c34da22cf796dba5640f87c04de5c95cf0adc11b85f1e1c1528-json.log",
+			Field:     "source",
+			Separator: "/",
+			Target:    "docker.container.id",
+			Index:     4,
+			Result:    "f1510836197d7c34da22cf796dba5640f87c04de5c95cf0adc11b85f1e1c1528",
+		},
+		{
+			Value:     "/var/lib/foo/bar",
+			Field:     "other_field",
+			Separator: "/",
+			Target:    "destination",
+			Index:     3,
+			Result:    "bar",
+		},
+		{
+			Value:     "-var-lib-foo-bar",
+			Field:     "source",
+			Separator: "-",
+			Target:    "destination",
+			Index:     2,
+			Result:    "foo",
+		},
+		{
+			Value:     "*var*lib*foo*bar",
+			Field:     "source",
+			Separator: "*",
+			Target:    "destination",
+			Index:     0,
+			Result:    "var",
+		},
+	}
+
+	for _, test := range tests {
+		var testConfig, _ = common.NewConfigFrom(map[string]interface{}{
+			"field":     test.Field,
+			"separator": test.Separator,
+			"index":     test.Index,
+			"target":    test.Target,
+		})
+
+		// Configure input to
+		input := common.MapStr{
+			test.Field: test.Value,
+		}
+
+		actual := runExtractField(t, testConfig, input)
+
+		result, err := actual.GetValue(test.Target)
+		assert.NoError(t, err)
+		assert.Equal(t, result.(string), test.Result)
```

Comment:
Does this type assertion panic if there was an err? I'm not a fan of assert.NoError due to it continuing on error. Maybe use "if assert.NoError { assert.Equal }". Check other uses and possibly replace with an "if err then t.Fatal".